samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,579 @@
+		ret = ota_wait_for_ip_config(ctx);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if (ctx->sock < 0) {
+		ret = ota_connect_to_server(ctx);
+		if (ret < 0) {
+			goto out_close;
+		}
+	}
+
+	ret = ota_download_firmware_chunk(ctx);
+	if (ret < 0) {
+		goto out_close;
+	}
+
+	ret = ota_load_firmware(ctx);
+	if (ret < 0) {
+		goto out_reset;
+	} else {
+		ret = ota_update_http_range_header(ctx);
+		if (ret < 0) {
+			goto out_reset;
+		}
+	}
+	return ret;
+out_close:
+	zsock_close(ctx->sock);
+	ctx->sock = -1;
+	ctx->flash_buffer_len = 0;
+	ctx->http_status_code = 0;
+	return ret;
+out_reset:
+	ctx->range_start = 0;
+	return ret;
+}
+
+int main(void)
+{
+	static struct app_ctx app_ctx = {
+		.events = Z_EVENT_INITIALIZER(app_ctx.events),
+		.sock = -1,
+		.range_start = 0,
+		.wifi_ssid = CONFIG_OTA_WIFI_SSID,
+		.wifi_psk = CONFIG_OTA_WIFI_PSK,
+		.security_type = WIFI_SECURITY_TYPE_PSK,
+		.connected = false,
+	};
+	const int max_retries = 2;
+	int retry_count = 0;
+	int ret = 0;
+
+	/* Register WiFi management callbacks */
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_wlan_scan_result, ota_handle_wifi_scan_result,
+				     NET_EVENT_WIFI_SCAN_RESULT);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_wlan_scan_result);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_wlan_scan_done, ota_handle_wifi_scan_done,
+				     NET_EVENT_WIFI_SCAN_DONE);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_wlan_scan_done);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_wlan_connect, ota_handle_wifi_connect_result,
+				     NET_EVENT_WIFI_CONNECT_RESULT);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_wlan_connect);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_wlan_disconnect,
+				     ota_handle_wifi_disconnect_result,
+				     NET_EVENT_WIFI_DISCONNECT_RESULT);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_wlan_disconnect);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_ipv4, ota_ipv4_event_handler,
+				     NET_EVENT_IPV4_ADDR_ADD | NET_EVENT_IPV4_ADDR_DEL);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_ipv4);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_ipv6, ota_ipv6_event_handler,
+				     NET_EVENT_IPV6_ADDR_ADD | NET_EVENT_IPV6_ADDR_DEL);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_ipv6);
+
+	net_mgmt_init_event_callback(&app_ctx.mgmt_cb_dns, ota_dns_event_handler,
+				     NET_EVENT_DNS_SERVER_ADD | NET_EVENT_DNS_SERVER_DEL);
+	net_mgmt_add_event_callback(&app_ctx.mgmt_cb_dns);
+
+	ota_parse_url(&app_ctx.url);
+	if (IS_ENABLED(CONFIG_NET_DHCPV4)) {
+		net_dhcpv4_start(net_if_get_first_wifi());
+	}
+	printf("OTA Application Started\n");
+	while (1) {
+		ret = ota_application_start(&app_ctx);
+		if (ret < 0) {
+			if (retry_count >= max_retries) {
+				printf("Maximum retries exceeded, aborting OTA\n");
+				break;
+			}
+			retry_count++;
+			printf("Error: %s, retrying (%d/%d)...\n", strerror(-ret), retry_count,
+			       max_retries);
+			k_sleep(K_MSEC(1000));
+		}
+	}
+	return (ret < 0) ? ret : 0;
```

Comment:
Are you sure you need to intricate 3 level of indentation for a such simple thing?

    do {
        ret = ota_application_start(&app_ctx);
        if (ret < 0) {
              retry_count++;
              k_sleep(K_MSEC(1000));
        }
    } while (retry_count < 3);
    printf("Maximum retries exceeded, aborting OTA\n");
    return -EIO;

(Note the detailed errors are already displayed in the earlier functions, having yet another error message here has little benefit).